README.md

```diff
@@ -24,60 +30,84 @@ To Install from the Nuget Package Manager Console
 
 Configuration and registration
 
-```c#
+<!-- snippet: ConfigurationAndRegistration -->
+```cs
 var sender = new MessageSender(connectionString, queueName);
 var config = new AzureStorageAttachmentConfiguration(storageConnectionString);
 sender.RegisterAzureStorageAttachmentPlugin(config);
-```        
+```
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L13-L19)</sup>
+<!-- endsnippet -->
 
 Sending
 
-```c#
-var payload = new MyMessage { ... }; 
+<!-- snippet: AttachmentSending -->
+```cs
+var payload = new MyMessage
+{
+    MyProperty = "The Value"
+};
 var serialized = JsonConvert.SerializeObject(payload);
 var payloadAsBytes = Encoding.UTF8.GetBytes(serialized);
 var message = new Message(payloadAsBytes);
 ```
-
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L24-L34)</sup>
+<!-- endsnippet -->
 
 Receiving
 
-```c#
+<!-- snippet: AttachmentReceiving -->
+```cs
 var receiver = new MessageReceiver(connectionString, entityPath, ReceiveMode.ReceiveAndDelete);
 receiver.RegisterAzureStorageAttachmentPlugin(config);
 var msg = await receiver.ReceiveAsync().ConfigureAwait(false);
 // msg will contain the original payload
 ```
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L39-L46)</sup>
+<!-- endsnippet -->
 
 ### Sending a message without exposing the storage account to receivers
 
 Configuration and registration with blob SAS URI
 
-```c#
+<!-- snippet: ConfigurationAndRegistrationSas -->
+```cs
 var sender = new MessageSender(connectionString, queueName);
 var config = new AzureStorageAttachmentConfiguration(storageConnectionString)
-	.WithBlobSasUri(sasTokenValidationTime: TimeSpan.FromHours(4), messagePropertyToIdentifySasUri: "mySasUriProperty");
+    .WithBlobSasUri(
+        sasTokenValidationTime: TimeSpan.FromHours(4),
+        messagePropertyToIdentifySasUri: "mySasUriProperty");
 sender.RegisterAzureStorageAttachmentPlugin(config);
-```  
+```
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L51-L60)</sup>
+<!-- endsnippet -->
 
 Sending
 
-```c#
```

Comment:
is this snippet meant to be the same as the above `AttachmentSending` one?

Reply:
Sending using connection string or SAS token looks the same. Could be merged I suppose 🙂